core/src/solution_submission.rs

```diff
@@ -174,77 +184,175 @@ impl<'a> StableXSolutionSubmitting for StableXSolutionSubmitter<'a> {
     }
```

Comment:
I think the easier solution would be to:
```rust
let mut submission = retry_with_gas_price_increase(...);
let mut deadline = wait_for_deadline(...);

select! {
    result = submission => result.map_err(self.make_error(...)),
    _ = b => {
        self.contract.send_noop_transaction(...);
        Err(...)
    },
};
```

Reply:
I thought about this. The disadvantage is that we don't know what gas price was used so we have to be maximally pessimistic. Since this hopefully happens rarely it is probably worth it for the simpler code. Will try and see how it looks.